src/tools/rustdoc-gui/tester.js

```diff
@@ -107,9 +107,11 @@ async function main(argv) {
     }
 
     files.sort();
+    let tests = [];
+    process.setMaxListeners(files.length + 1);
```

Comment:
This improvement is super cool, however I see a few potential problems:

 * The tests are not alphabetically sorted anymore (I guess it's not too hard to fix though...).
 * The error output is kind of a mess now since it doesn't relate to its test anymore.
 * I'm not sure how well it'll work out on the CI (didn't seem to encounted a problem, so I think it's fine).

I think it's a bit more tricky that it might seem, so maybe put it into another PR so we can investigate with peace in mind?

Reply:
Oh also: you can run one specific test by passing its name: `x.py test src/test/rustdoc-gui/test.goml` (if that helps you?).

Reply:
Actually, fixing the alphabetical order is pretty hard. So, I will implement something that is more readable